x-pack/plugins/ml/public/embeddables/types.ts

```diff
@@ -133,6 +135,20 @@ export interface SingleMetricViewerEmbeddableCustomInput {
 export type SingleMetricViewerEmbeddableInput = EmbeddableInput &
   SingleMetricViewerEmbeddableCustomInput;
 
+export interface SingleMetricViewerFieldSelectionApi {
+  input: Partial<SingleMetricViewerEmbeddableInput>;
+}
+
+export interface SingleMetricViewerComponentApi extends SingleMetricViewerFieldSelectionApi {
+  updateUserInput: (input: Partial<SingleMetricViewerEmbeddableInput>) => void;
+}
```

Comment:
It seems like API for the SMV embeddable should look like this. 

```suggestion
export interface SingleMetricViewerComponentApi {
  // You should flatten the `input` here with relevant properties
  jobIds: JobId[];
  selectedDetectorIndex: number;
  selectedEntities?: MlEntityField[];
  
  updateUserInput: (input: Partial<SingleMetricViewerEmbeddableInput>) => void;
}
```

I also think that `functionDescription` is redundant and should not be stored in the embeddable state, because you can retrieve it from the analysis config knowing the `detectorIndex`.

Reply:
Updated in bc6fb2ca9b1833fffc6f524af3055ef317a18b47